### PR TITLE
Feature/629 state filter by condition

### PR DESCRIPTION
**Pull request by cschwinderg**
## Related Issues:
* [HMW-629](https://jira.epa.gov/browse/HMW-629)

## Main Changes:
* Added a "Filter Waterbodies By" control to the state advanced search page.

## Steps To Test:
1. Navigate to http://localhost:3000/state/HI/advanced-search
2. Select "Algae" and "Bacteria and Other Microbes" in the "Parameter Groups" field
3. Select "Swimming and Boating" in the "Use Groups" field
4. Click Search
5. Test the new "Filter Waterbodies by" control including changing the "Display Waterbodies by" option

### CODE REVIEW

**Review by maxdiebold-erg — CHANGES_REQUESTED** [review]
Great job, I think I prefer the look of the updated layout. Only one suggestion: all other inputs are reset when changing location, so the `Filter Waterbodies by` input should probably be reset, too.

**Comment by cschwinderg** [comment]
Good call! I'll make that update shortly.

File: app/client/src/components/pages/StateTribal.Tabs.AdvancedSearch.js [hunk 1]
```diff
@@ -949,6 +978,7 @@ function AdvancedSearch() {
             // set the display back to the default option
```
Comment (maxdiebold-erg) [inline app/client/src/components/pages/StateTribal.Tabs.AdvancedSearch.js:978]: This was an issue before this PR, but I only saw it while testing the below addition: it seems like the intention of the `indexOfDisplay` is to keep a selection across searches if it's in both. However, the groups used in `react-select` result in a nested array and the `indexOfDisplay` is only ever found for "Overall Waterbody Condition". The above lines 975-977 could be changed to:
```
const indexOfDisplay = newDisplayOptions
  .flatMap((item) => item.options ?? item)
  .findIndex((item) => item.value === selectedDisplayOption.value);
```

**Review by maxdiebold-erg — CHANGES_REQUESTED** [review]
I added one more suggestion for an unrelated change.

File: app/client/src/components/pages/StateTribal.Tabs.AdvancedSearch.js (hunk 1 above)
Comment (cschwinderg) [reply app/client/src/components/pages/StateTribal.Tabs.AdvancedSearch.js:979]: Good catch! I think that select used to not have the groupings.

**Review by maxdiebold-erg — APPROVED** [review]

**Pull request merged**